src/main/java/org/betonquest/betonquest/modules/schedule/EventScheduling.java

```diff
@@ -0,0 +1,144 @@
+import org.betonquest.betonquest.exceptions.InstructionParseException;
+import org.betonquest.betonquest.exceptions.ObjectNotFoundException;
+import org.betonquest.betonquest.id.ScheduleID;
+import org.bukkit.configuration.ConfigurationSection;
+
+import java.lang.reflect.InvocationTargetException;
+import java.util.Map;
+import java.util.Optional;
+
+/**
+ * Class responsible for managing schedule types, their schedulers, as well as parsing schedules from config.
+ */
+@CustomLog(topic = "Schedules")
+public class EventScheduling {
+
+    /**
+     * Map that contains all types of schedulers,
+     * with keys being their names and values holding the scheduler and schedule class.
+     */
+    private final Map<String, ScheduleType<?>> scheduleTypes;
+
+    /**
+     * Creates a new instance of the event scheduling class.
+     *
+     * @param scheduleTypes map containing the schedule types, provided by {@link org.betonquest.betonquest.BetonQuest}
+     */
+    public EventScheduling(final Map<String, ScheduleType<?>> scheduleTypes) {
+        this.scheduleTypes = scheduleTypes;
+    }
+
+    /**
+     * Method used for loading all schedules of a quest package and registering them in the correct schedulers.
+     *
+     * @param questPackage package to load
+     */
+    public void loadData(final QuestPackage questPackage) {
+        LOG.debug(questPackage, "Parsing schedules for package '" + questPackage.getPackagePath() + "'.");
+        final ConfigurationSection configuration = questPackage.getConfig().getConfigurationSection("schedules");
+        if (configuration == null) {
+            LOG.debug(questPackage, "Package contains no schedules.");
+            return;
+        }
+        int count = 0;
+        for (final String key : configuration.getKeys(false)) {
+            if (key.contains(" ")) {
+                LOG.warn(questPackage,
+                        "Schedule name cannot contain spaces: '" + key + "' (in " + questPackage + " package)");
+                continue;
+            }
+
+            try {
+                final ScheduleID scheduleID = new ScheduleID(questPackage, key);
+                try {
+                    final ConfigurationSection scheduleConfig = new UnmodifiableConfigurationSection(
+                            questPackage.getConfig().getConfigurationSection("schedules." + scheduleID.getBaseID())
+                    );
+                    final String type = Optional.ofNullable(scheduleConfig.getString("type"))
+                            .orElseThrow(() -> new InstructionParseException("Missing type instruction"));
+                    final ScheduleType<?> scheduleType = Optional.ofNullable(scheduleTypes.get(type))
+                            .orElseThrow(() -> new InstructionParseException("The schedule type '" + type + "' is not defined"));
+                    scheduleType.createAndScheduleNewInstance(scheduleID, scheduleConfig);
+                    count++;
```

Comment:
For some reason, i dont like the counter here, only to show the debug message. There is no other solution? I am not sure if we even need the debug message for the amount, because we already have an debug message for every successful loaded schedule